app/scripts/controllers/permissions/background-api.js

```diff
@@ -38,23 +43,39 @@ export function getPermissionBackgroundApiMethods({
   };
 
   // To add more than one account when already connected to the dapp
-  const addMoreAccounts = (origin, accounts) => {
+  const addMoreAccounts = (origin, addresses) => {
     const caip25Caveat = getCaip25Caveat(origin);
     if (!caip25Caveat) {
       throw new Error(
         `Cannot add account permissions for origin "${origin}": no permission currently exists for this origin.`,
       );
     }
 
-    const ethAccounts = getEthAccounts(caip25Caveat.value);
+    const internalAccounts = addresses.map((address) => {
+      return accountsController.getAccountByAddress(address);
```

Comment:
This is gonna result in worse perf than we had before for this logic

Reply:
🤷 yes it will. I'm not sure there's a good way around it since we need to map back internalAccounts to the addresses at some point?

Reply:
the only way around this is to start passing caipAccountId around everywhere in component state, and this is simple too much to handle on top of everything else

Reply:
Account menu account selection already has been using this address overloading hack

Reply:
Yeah, fair!